### PR TITLE
fix(fuzz): Reduce false positives in fuzz-testing

### PR DESCRIPTION
**Pull request by fpbgg**
Supersedes #3438

### CODE REVIEW

File: clarity/src/vm/types/signatures.rs [hunk 1]
```diff
@@ -1180,26 +1180,63 @@ impl TypeSignature {
...
+    }
+
+    // This function checks recursively if NoType is anywhere in the type signature.
+    #[cfg(feature = "fuzzing")]
+    pub fn contains_invalid_type_rhs(ty: &TypeSignature) -> bool {
+        match ty {
+            NoType => true,
+            OptionalType(inner_type) => {
+                match inner_type.as_ref() {
+                    NoType => true, // NoType is a valid inner type for OptionalType
```
Comment (obycode) [inline clarity/src/vm/types/signatures.rs:1226]: I thought that for the rhs, a `NoType` is valid here (like the comment says), so shouldn't this be `false`?

File: clarity/src/vm/types/signatures.rs [hunk 2]
```diff
@@ -1180,26 +1180,63 @@ impl TypeSignature {
...
+                    inner_type => Self::contains_invalid_type_rhs(inner_type),
+                }
+            }
+            ResponseType(inner_types) => {
+                Self::contains_invalid_type_rhs(&inner_types.0)
+                    || Self::contains_invalid_type_rhs(&inner_types.1)
```
Comment (obycode) [inline clarity/src/vm/types/signatures.rs:1232]: Similarly here, I think it should only be invalid if both the ok-type and the err-type are invalid types.

File: clarity/src/vm/types/signatures.rs (hunk 1 above)
Comment (fpbgg) [reply clarity/src/vm/types/signatures.rs:1226]: Good catch. I think this was an error in recovering from the git rebase.

**Review by obycode — APPROVED** [review]
Thanks @fpbgg!

**Pull request merged**